src/Adapter/Order/CommandHandler/DeleteProductFromOrderHandler.php

```diff
@@ -70,6 +100,49 @@ public function handle(DeleteProductFromOrderCommand $command)
         $order = $order->refreshShippingCost();
 
         Hook::exec('actionOrderEdited', ['order' => $order]);
+
+        $this->contextStateManager->restoreContext();
+    }
+
+    /**
+     * @param Order $order
+     * @param OrderDetail $orderDetail
+     * @param Cart $cart
+     *
+     * @return bool
+     */
+    private function updateCart(Order $order, OrderDetail $orderDetail, Cart &$cart)
+    {
+        $oldCartRules = $cart->getCartRules();
+        // Update Product Quantity in the cart
+        $return = $cart->updateQty(
+            $orderDetail->product_quantity,
+            $orderDetail->product_id,
+            $orderDetail->product_attribute_id,
+            false,
+            'down'
+        );
+        // Remove & Add CartRules applied to cart
+        CartRule::autoRemoveFromCart();
+        CartRule::autoAddToCart();
+
+        // Remove previous cart rules applied to order
+        $result = array_diff(
+            array_map('serialize', $oldCartRules),
+            array_map('serialize', $cart->getCartRules())
+        );
+        $result = array_map('unserialize', $result);
+
+        foreach ($order->getCartRules() as $orderCartRule) {
+            foreach ($result as $itemCartRule) {
+                if ($itemCartRule['id_cart_rule'] == $orderCartRule['id_cart_rule']) {
+                    $orderCartRule = new OrderCartRule($orderCartRule['id_order_cart_rule']);
+                    $orderCartRule->delete();
+                }
```

Comment:
I'm not sure I understand what is being done here (the whole foreach), I'm not saying it's wrong, I just don't understand/know what you are doing exactly and why ?

Reply:
;)